src/json/from_json.cr

```diff
@@ -65,6 +65,53 @@ def Deque.from_json(string_or_io) : Nil
   end
 end
 
+module Iterator(T)
+  # Reads the content of a JSON array into an iterator in a lazy way.
+  # With this method it should be possible to process a huge JSON array, without
+  # the requirement that the whole array fits into memory.
+  #
+  # The following example produces a huge file, uses a lot of CPU but should not require much memory.
+  #
+  # ```
+  # File.open("/tmp/test.json", "w+") do |f|
+  #   (0..1_000_000_000).each.to_json(f)
+  # end
+  #
+  # File.open("/tmp/test.json", "r") do |f|
+  #   p Iterator(Int32).from_json(f).skip(1_000_000_000).to_a
+  # end
+  # ```
+  def self.from_json(string_or_io)
```

Comment:
It would be nice to add a warning that the IO/PullParser can't be used by anything else until the iterator is fully consumed.

Reply:
I added a warning.